python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -318,21 +316,61 @@ class StreamingExecutor:
     with the 'distributed' scheduler.
     """
 
+    _env_prefix = "CUDF_POLARS__EXECUTOR"
+
     name: Literal["streaming"] = dataclasses.field(default="streaming", init=False)
-    scheduler: Scheduler = Scheduler.SYNCHRONOUS
+    scheduler: Scheduler = dataclasses.field(
+        default_factory=_make_default_factory(
+            f"{_env_prefix}__SCHEDULER", Scheduler.__call__, Scheduler.SYNCHRONOUS
```

Comment:
(Non-blocking) For the sake of clarity, I think I like the defaults being passed with the `default=` keyword. Just a note - Up to you.